process_report/invoices/billable_invoice.py

```diff
@@ -0,0 +1,186 @@
+from dataclasses import dataclass
+from decimal import Decimal
+import logging
+import sys
+
+import pandas
+import pyarrow
+
+import process_report.invoices.invoice as invoice
+import process_report.util as util
+
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+@dataclass
+class BillableInvoice(invoice.Invoice):
+    nonbillable_pis: list[str]
+    nonbillable_projects: list[str]
+    old_pi_filepath: str
+
+    def _prepare(self):
```

Comment:
Should this operate on a copy of the data? Or is a copy of the dataframe returned from `_remove_nonbillables`? I'm not very familiar with pandas so this is a bit of a basic question.

Reply:
When passed into a function, mutable objects like dataframes behave as if they're "passed-by-reference". Which is to say that unless I explicitly pass in a copy of the dataframe, every function could and will change the passed-in dataframe. 

I suppose this makes the return statements in most of the dataframe processing functions redundant. I can remove them at a later PR if you'd like.

Reply:
@QuanMPhm if you are happy with your response please resolve the conversation.